lib/functions/process-message.js

```diff
@@ -33,6 +35,25 @@ module.exports.handler = async (event, context) => {
     // close database pool
     await db.end()
 
+    // Invoke lambda function to store warning alerts in S3 bucket
+
+    const lambda = new AWS.Lambda({
+      endpoint: new AWS.Endpoint(process.env.FWS_LAMBDA_ENDPOINT || 'localhost')
+    })
+
+    const params = {
+      FunctionName: (process.env.FWS_ENV_NAME || 'dev') + 'fws-putWarningsAlerts',
+      InvocationType: 'Event',
+      Payload: JSON.stringify({ TEST: 'Test' }) // TODO: fix object
+    }
+
+    await lambda.invoke(params, (err, data) => {
```

Comment:
@JFSCH unsure about this, there is the await keyword, but what is it awaiting as it doesn't return a promise, and simply uses the callback to log the data.

Should we not either use `const response = await lambda.invoke(params).promise()` to return the promise  or just take out the await?

Reply:
@teddmason Introduced changes as requested and tested invoking function via app and aws cli. However,  build not passing due to failing unit test (although all tests pass locally) so will investigate. Additionally, testing highlighted that new function is not being invoked when 'cold'. Can rectify this with serverless-plugin-warmup but unsure if this is ideal approach. Researching alternatives if warmup plugin deemed unsuitable.

Reply:
Is it failing to invoke the cold function after this promise change, or was it doing so with the callback? If not just return it to what it was then, otherwise why is it failing to wait for the lambda function to warm up and execute? Is the calling function timing out, if so could we update the timeout value. I would be against running anything to keep the function warm, would be a bit anti-pattern

Reply:
@teddmason Code refactored and unit tests updated to stub lambda invoke so build now passes. Code now works for invoking function when cold. Did not have to increase timeout or introduce warmup plugin.